### ISSUE TITLE
Module imported with `use` at function scope not available to nested `try` block

### ISSUE THREAD

**Issue by krader1961** [elves/elvish#495]
This is the test case:

```
fn wtf {
    use re
    try {
        e:false
    } except e {
        try {
            local:matches = (re:find ' exited with (\d+)''' (echo $e))
            print $matches[groups][1][text]
        } except e2 {
            put 're:find failed' $e2
        }
    }
}
```

When I run that function it throws `?(fail 'search re:find: not found')`. I have to move the `use` to within the innermost `except { }` block for it to work. That seems wrong. It doesn't matter if I `use re` at the top-most scope. Shouldn't it have walked up the enclosing scopes till it found that the module had been imported and only fail in that manner if it hadn't been imported?

**Comment by krader1961** [comment]
I feel like this is expected behavior and I have misunderstood something fundamental about how module references are resolved since this simpler example fails in the same way:

```
fn wtf {
    use re
    { re.find 'hello' 'abc hello def' }
}
```

But it is still counterintuitive is not wrong.

**Comment by xiaq** [comment]
Thanks for reporting! What you described is a bug. Module imports should be lexically scoped, like variables. (Lexical scoping of module imports was only introduced recently, it seems I should introduce some tests for that.)

**Comment by xiaq** [comment]
Interestingly, this seems to only affect function calls. The following works fine:

```
~> cat .elvish/lib/lorem.elv 
name = lorem
fn ipsum {
  echo lorem ipsum
}
~> fn f {
     use lorem
     { echo $lorem:name }
   }
~> f
lorem
```

This doesn't:

```
~> fn f {
     use lorem
     { lorem:ipsum }
   }
~> f
Exception: search lorem:ipsum: not found
Traceback:
  [interactive], line 3:
      { lorem:ipsum }
  [interactive], line 3:
      { lorem:ipsum }
  [interactive], line 1:
    f
```

**Comment by xiaq** [comment]
Fixed in 904eb11a64d5687924a93928baddae1f8332f833.
